pkg/gomod/selector.go

```diff
@@ -15,90 +14,49 @@ var (
 )
 
 // Matcher matches a module name.
-type Matcher interface {
```

Comment:
I would really like to keep at least this interface.

It allows people passing not only functions, but compatible structs as well.

It can be easily implemented by simple func:

```go
type MatcherFn func(modname string) bool
```

and to return it:

```go
return MatcherFn(func(modname string) bool {
  return strings.HasPrefix(modname, domain) && !currentModules(modname)
})
```

Reply:
But, if you don't feel it is worth it, I can give LGTM right away.

Reply:
Did you see how I used `StringSet.Has` as a `Matcher` below on line 41?

```golang
func CurrentModulesMatcher() (Matcher, error) {
  knownMods := sets.NewString()
  // ...
  return knownMods.Has, nil
}
```

I'm not convinced we need to declare an `interface` to define an interface (`Matcher` as a type alias for `func(string) bool`).

Reply:
So in your example above, you could have:

```golang
  return MatcherFn(myMatcher).Match
```

Reply:
/lgtm
/approve